pkg/tsdb/cloudmonitoring/cloudmonitoring.go

```diff
@@ -66,145 +72,205 @@ const (
 	perSeriesAlignerDefault   string = "ALIGN_MEAN"
 )
 
-func ProvideService(cfg *setting.Cfg, pluginManager plugins.Manager, httpClientProvider httpclient.Provider) *Service {
-	return &Service{
-		PluginManager:      pluginManager,
-		HTTPClientProvider: httpClientProvider,
-		Cfg:                cfg,
+func ProvideService(cfg *setting.Cfg, httpClientProvider httpclient.Provider, pluginManager plugins.Manager,
+	backendPluginManager backendplugin.Manager, dataSourceCache datasources.CacheService) *Service {
+	s := &Service{
+		pluginManager:        pluginManager,
+		backendPluginManager: backendPluginManager,
+		httpClientProvider:   httpClientProvider,
+		dataSourceCache:      dataSourceCache,
+		cfg:                  cfg,
+		im:                   datasource.NewInstanceManager(newInstanceSettings(httpClientProvider)),
 	}
+
+	factory := coreplugin.New(backend.ServeOpts{
+		QueryDataHandler: s,
+	})
+
+	if err := s.backendPluginManager.Register("stackdriver", factory); err != nil {
+		slog.Error("Failed to register plugin", "error", err)
+	}
+	return s
 }
 
 type Service struct {
-	PluginManager      plugins.Manager
-	HTTPClientProvider httpclient.Provider
-	Cfg                *setting.Cfg
+	pluginManager        plugins.Manager
+	backendPluginManager backendplugin.Manager
+	dataSourceCache      datasources.CacheService
+	httpClientProvider   httpclient.Provider
```

Comment:
i think this is not needed to be stored in service anymore, it was there just because of dependency injection i think. now with newInstanceSettings() we got the http client and store it in datasourceinfo, then use the client all the time.